nikola/data/themes/base/templates/feeds_translations_helper.tmpl

```diff
@@ -1,91 +1,97 @@
+            % for language in sorted(translations):
+                % if classification and kind != 'index':
+                    <link rel="alternate" type="application/rss+xml" title="RSS for ${kind} ${classification|h}${_append_language(language)}" hreflang="${language}" href="${_link(kind + "_rss", classification, language)}">
+                % else:
+                    <link rel="alternate" type="application/rss+xml" title="RSS${_append_language(language)}" hreflang="${language}" href="${_link("index_rss", None, language)}">
                 % endif
-            % endif
-        % endfor
-    % else:
-        % if classification:
-            % if generate_atom:
-                <link rel="alternate" type="application/atom+xml" title="Atom for ${kind} ${classification|h}" href="${_link(kind + "_atom", classification)}">
-            % endif
-            % if generate_rss and not rss_link:
-                <link rel="alternate" type="application/rss+xml" title="RSS for ${kind} ${classification|h}" href="${_link(kind + "_rss", classification)}">
-            % endif
+            % endfor
+        % endif
+    % endif
+</%def>
+
+<%def name="_head_atom(classification=None, kind='index')">
+    % if generate_atom:
+        % if len(translations) > 1 and has_other_languages and classification and kind != 'index':
+            % for language, classification, name in all_languages:
+                <link rel="alternate" type="application/atom+xml" title="Atom for ${kind} ${name|h} (${language})" hreflang="${language}" href="${_link(kind + "_atom", classification, language)}">
+            % endfor
         % else:
-            % if generate_atom:
-                <link rel="alternate" type="application/atom+xml" title="Atom" href="${_link("index_atom", None)}">
-            % endif
-            % if generate_rss and not rss_link:
-                <link rel="alternate" type="application/rss+xml" title="RSS" href="${_link("rss", None)}">
-            % endif
+            % for language in sorted(translations):
+                % if classification and kind != 'index':
+                    <link rel="alternate" type="application/atom+xml" title="Atom for ${kind} ${classification|h}${_append_language(language)}" hreflang="${language}" href="${_link(kind + "_atom", classification, language)}">
+                % else:
+                    <link rel="alternate" type="application/atom+xml" title="Atom${_append_language(language)}" hreflang="${language}" href="${_link("index_atom", None, language)}">
+                % endif
+            % endfor
         % endif
     % endif
-    % if has_other_languages and other_languages:
+</%def>
+
+## Handles both feeds and translations
+<%def name="head(classification=None, kind='index', feeds=True, other=True, rss_override=True, has_no_feeds=False)">
+    % if feeds and not has_no_feeds:
+        ${_head_rss(classification, 'index' if (kind == 'archive' and rss_override) else kind, rss_override)}
```

Comment:
That survived jinjification (as opposed to the above) and should work fine in jinja.